File: transcoder/src/handlers/user/validators.rs
pub mod validators {
    pub fn validate_email(email: &str) -> Result<(), String> {
        if email.contains('@') && email.contains('.') && email.len() > 5 {
            Ok(())
        } else {
            Err("Invalid email format".to_string())
        }
    }

    pub fn validate_password(password: &str) -> Result<(), String> {
        if password.len() < 8 {
            return Err("Password must be at least 8 characters".to_string());
        }
        if !password.chars().any(|c| c.is_uppercase()) {
            return Err("Password must contain at least one uppercase letter".to_string());
        }
        if !password.chars().any(|c| c.is_digit(10)) {
            return Err("Password must contain at least one number".to_string());
        }
        Ok(())
    }
}

File: web/src/utils/validators.js
/**
 * Validate email format
 * @param {string} email - Email to validate
 * @returns {boolean} True if valid
 */
export const validateEmail = (email) => {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return emailRegex.test(email);
};

/**
 * Validate password strength
 * @param {string} password - Password to validate
 * @returns {Object} Validation result with score and message
 */
export const validatePassword = (password) => {
  let score = 0;
  const feedback = [];

  if (password.length >= 8) score++;
  else feedback.push("At least 8 characters");

  if (/[A-Z]/.test(password)) score++;
  else feedback.push("One uppercase letter");

  if (/[a-z]/.test(password)) score++;
  else feedback.push("One lowercase letter");

  if (/[0-9]/.test(password)) score++;
  else feedback.push("One number");

  if (/[!@#$%^&*]/.test(password)) score++;
  else feedback.push("One special character (!@#$%^&*)");

  return {
    score,
    isValid: score >= 3,
    feedback:
      feedback.length > 0 ? `Add: ${feedback.join(", ")}` : "Strong password",
  };
};

/**
 * Validate required field
 * @param {any} value - Value to validate
 * @returns {boolean} True if not empty
 */
export const isRequired = (value) => {
  return value !== null && value !== undefined && value !== "";
};
